### DISCUSSION TITLE
Make script language tag optional.

### DISCUSSION THREAD

**Discussion by wapoa** [massCodeIO/massCode#178] [Ideas]
Hi there, loving the tool. 

I like the syntax coloring, but I am wondering why do I still need a <?php tag  if the language syntax is already set to PHP.  I think that should be a setting to show language tags or not. 

In that vein if we had a quick copy snippet button then it would be easier to just get that small snippet in my code. If the snippet would have the tags in them it would be annoying to always have to remove the tag when pasting in the code.

So in short
 * syntax color coding without needing script tags
 * quick snippet copy button (maybe there's already a way to do this and I didn't figure it out?)

Thanks!

**Comment by antonreshetov** [comment]
massCode uses Ace as an editor without modifying it. That's why everything about syntax highlighting is on it. It seems that the tokenizer for php needs the `<?php` tag to work correctly.

**Comment by antonreshetov** [comment]
Initially the `<>` button was responsible for copying the snippet, but later with the advent of HTML & CSS rendering this button became responsible for displaying the preview. I think it makes more sense now. 

You can quickly copy a snippet by simply pressing `cmd + shift + c`

**Comment by antonreshetov** [comment]
I found a solution to make Ace highlight without `<?php`. I will add it in the next version. 
![image](https://user-images.githubusercontent.com/17480987/181683039-f24f51da-3a98-4d2b-9d2a-cfbf22be786f.png)

**Comment by wapoa** [comment]
That's awesome thanks!

**Reply by antonreshetov** [reply]
@wapoa see https://github.com/massCodeIO/massCode/discussions/132#discussioncomment-3331639

**Reply by wapoa** [reply]
Awesome thanks will be trying it out soon!
